andela/ah-backend-realers
# #166187866 Receiving Notications

#### What does this PR do?

Enables users to receive notifications whenever authors they follow create new articles
Enables users to receive notifications whenever someone comments on an article they favorited

#### Description of task completed
* Updated the `ArticleView` class `post` method

#### How should this be manually tested on your local machine
* Make sure you [postman](https://www.getpostman.com/downloads/) installed on your local machine
* Do a `git fetch origin ft-receive-notifications-166187866`
* Do a `git checkout ft-receive-notifications-166187866`
* Run this command `$ python manage.py runserver`
* The url provided in your terminal should be tested on postman with the prefixes below
 `/api/users/` for `POST` method when the user is registering.

use the format below
`{
       "user":{
              "email": "email",
              "username": "username",
              "password": "password"
         }
               }`
* Register 3 users for demonstration purposes and make sure the accounts are activated
* Then login on postman with the prefixes below
`/api/users/login/` for `POST` method.

use the format below
`{
       "user":{
              "email": "email",
               "password": "password"
         }
               }`
* After logging in, you should have received the token
* Using the follow users endpoint get 2 users of your choice to follow a single user.
* Login with an account of a single user followed by the two users above
* Create an article using postman with the prefixes below with the token in the authorization header
`/api/articles/ on a `POST` method. use the format below

`{
	"article":{
		"title": "title",
		"description": "description",
		"body": "body"
	}
}`

* Check the respective emails for the two users following this user who has created an article.
* The two users should have received emails notifying of the user they follow having created a new article plus its title.
finishes [166187866](https://www.pivotaltracker.com/n/projects/2350340/stories/166187866)
